src/material-experimental/mdc-slider/slider-adapter.ts

```diff
@@ -6,133 +6,179 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
+import {ChangeDetectorRef, ElementRef, Inject} from '@angular/core';
 import {SpecificEventListener, EventType} from '@material/base';
 import {MDCSliderAdapter, Thumb, TickMark} from '@material/slider';
+import {MatSliderThumb, MAT_SLIDER} from './slider-thumb';
 
+/**
+ * This is a dummy interface that just contains the properties and methods of MatSlider that are
+ * used by SliderAdapter. Rather than directly referencing MatSlider, we use this interface when
+ * to avoid a circular dependency between MatSlider and SliderAdapter.
+ */
+interface MatSlider {
```

Comment:
Shouldn't the slider implement this interface? Otherwise there's nothing keeping the two in sync.

Reply:
If we use `MatSlider` directly instead of the interface, won't we still get circular deps?

Reply:
Kristiyan means that you should have e.g.
```ts
class MatSlider implements SliderThumbContainer {
}
```